### PR TITLE
182235395 user watchlist

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
- users should be able to add validators to watchlist

#### How should this be manually tested?
- log in
- you should be able to see the eye under the avatars on the main page - it adds validator to watchlist
- to filter by watched validators click the eye on the top of the table (over the avatars column)
- run tests

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/182235395)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [Y] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

**Review by radek-polcode — CHANGES_REQUESTED** [review]

File: test/factories/user_watchlist_elements.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user_watchlist_element do
```
Comment (radek-polcode) [inline test/factories/user_watchlist_elements.rb:4]: It's a useless factory where everything is nil I think

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def current_user_info
+    response = if current_user
+                { api_token: current_user.api_token }
+              else
```
Comment (radek-polcode) [inline app/controllers/users_controller.rb:7]: I think it's still not aligned correctly

File: app/queries/validator_query.rb [hunk 3]
```diff
@@ -2,15 +2,25 @@
...
-    @default_scope = Validator.select(validator_fields, validator_score_v1_fields)
-                              .joins(:validator_score_v1_for_api)
-                              .includes(
-                                :vote_accounts_for_api,
-                                :most_recent_epoch_credits_by_account,
-                                validator_ip_active_for_api: [data_center_host_for_api: [:data_center_for_api]]
-                              )
-                              
+  def initialize(user_id: nil)
+    if user_id
```
Comment (radek-polcode) [inline app/queries/validator_query.rb:6]:
```suggestion
    @default_scope = if user_id
```

File: app/queries/validator_query.rb [hunk 4]
```diff
@@ -2,15 +2,25 @@
...
+      user = User.find(user_id)
+      @default_scope = user.watched_validators.select(validator_fields, validator_score_v1_fields)
+                                .joins(:validator_score_v1_for_api)
```
Comment (kbiala) [inline app/queries/validator_query.rb:9]: Fix indentation

File: app/queries/validator_query.rb (hunk 3 above)
Comment (kbiala) [inline app/queries/validator_query.rb:6]: It could have been one line:
if user_id && User.find(user_id)

File: app/models/validator.rb [hunk 5]
```diff
@@ -47,6 +47,10 @@ class Validator < ApplicationRecord
   has_many :validator_block_histories, dependent: :destroy
   has_many :commission_histories, dependent: :destroy
   has_many :validator_histories, primary_key: :account, foreign_key: :account
+
+  has_many :user_watchlist_elements
```
Comment (radek-polcode) [inline app/models/validator.rb:51]:
```suggestion
  has_many :user_watchlist_elements, dependent: :destroy
```

File: app/models/user.rb [hunk 6]
```diff
@@ -52,6 +52,9 @@ class User < ApplicationRecord
          :confirmable, :lockable, :timeoutable, :trackable,
          authentication_keys: [:username]
 
+  has_many :user_watchlist_elements
```
Comment (radek-polcode) [inline app/models/user.rb:55]:
```suggestion
  has_many :user_watchlist_elements, dependent: :destroy
```

File: db/schema.rb [hunk 7]
```diff
@@ -176,66 +176,6 @@
     t.index ["network", "epoch"], name: "index_epoch_wall_clocks_on_network_and_epoch", unique: true
   end
 
-  create_table "ip_overrides", charset: "utf8mb4", collation: "utf8mb4_0900_ai_ci", force: :cascade do |t|
```
Comment (kbiala) [inline db/schema.rb:179]: These changes are not related to this task

**Review by kbiala — CHANGES_REQUESTED** [review]

**Review by radek-polcode — CHANGES_REQUESTED** [review]

**Pull request merged**